feat: use placeholder instead image

ferrarafer opened the pull request:
- Replace real images with local placeholders
- Stall the app until session selected
- Improved data masking

NOTE: When I was wrapping the changes, I exceeded book api quota 😄 , I will double check all tomorrow, maybe will be better to avoid merge till tomorrow.

> Exception: Quota exceeded for quota metric 'Queries' and limit 'Queries per day' of service 'books.googleapis.com' for consumer.


**DEMO**

https://github.com/FilledStacks/session_mate/assets/1505545/78886fd0-2083-4b7b-ba36-acf7e4531d35

https://github.com/FilledStacks/session_mate/assets/1505545/988934c8-cbd0-4b8f-897c-d0f046cbb516

ferrarafer committed:
fix: update example app

- Throw exception for status code 429
- Show the index instead of bookId on BookCard
- Use environment variable to use fake data

ferrarafer committed:
chore: add placeholder image

Not final image, needs to be changed for a better one.

ferrarafer committed:
fix: update http classes to improve data masking

ferrarafer committed:
refactor: improve hashEvent

ferrarafer committed:
fix: update _stringSubstitution

Fix _stringSubstitution to avoid one extra character per space in the String.

ferrarafer committed:
feat: add more environment variables

ferrarafer committed:
fix: avoid record a session on replay mode

ferrarafer committed:
feat: add local placeholders

ferrarafer committed:
feat: improve session services for data masking

ferrarafer committed:
feat: use a Builder to avoid guest app before session selected

FilledStacks reviewed (changes requested):
This looks really great! I'm happy with it. 

One thing we need to do is to stall the app until we tap the play button, not until we select the session. 

Reason: we only need to fill the cache at that point since we might be tapping through different sessions before we select one to play.

ferrarafer committed:
fix: dispose form on BookDetailsView dispose

ferrarafer committed:
feat: add ConfigurationService

ferrarafer committed:
refactor: SessionMate and GuestRestarter

- Changed SessionMate to SessionMateBuilder
- Converted SessionMateBuilder from StatelessWidget to StackedView
- Changed GuestRestarter to SessionMate

ferrarafer committed:
feat: trigger restart as callback

Set a callback on DriverCommunicationService to trigger the restart of the app when the replay finishes.

ferrarafer committed:
fix: remove restart button from DriverBar

ferrarafer committed:
feat: stall the app until session is replayed

ferrarafer committed:
fix: update DataMaskingService

ferrarafer commented:
**New Demo**

https://github.com/FilledStacks/session_mate/assets/1505545/9c8df236-e178-4ae3-9075-3b1f63abf72c

FilledStacks commented on example/lib/main.dart:
```diff
@@ -26,11 +26,18 @@ class MainApp extends StatelessWidget {
       ],
-      builder: (context, child) => SessionMate(
+      builder: (context, child) => SessionMateBuilder(
+        excludeKeysOnDataMasking: const [
```
Perfect

FilledStacks commented on lib/src/widgets/driver_ui/driver_ui.dart:
```diff
@@ -9,8 +10,7 @@ import 'package:stacked/stacked.dart';
   final Widget child;
-  final VoidCallback? onRestart;
-  const DriverUI({super.key, required this.child, this.onRestart});
+  const DriverUI({super.key, required this.child});
```
I'm very happy to see the required restart code removed

FilledStacks commented on test/services/data_masking_service_test.dart:
🔥

FilledStacks commented on test/services/configuration_service_test.dart:
🔥

FilledStacks reviewed (approved):
Great PR! Local session replay is wrapped up nicely with these features.
